### ISSUE TITLE
Bug: Canonical logic accept domain.com/////

### ISSUE THREAD

**Issue by janbarasek** [nette/routing#6]
- bug report? yes
- feature request? yes
- version: `v3.0.0`

### Description

In case of `domain.com////` internal method `match(IRequest $httpRequest): ?array` got fake URL with only one slash.

Some example (`CustomRouter.php`):

```php
public function match(IRequest $httpRequest): ?array
{
	$url = $httpRequest->getUrl();
	dump($url);
	dump($url->getAbsoluteUrl());
	$realUrl = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? 'https' : 'http')
		. '://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
	dump($realUrl);
	die;
```

In production mode dump looks like this:

![Snímek obrazovky 2020-02-12 v 10 05 47](https://user-images.githubusercontent.com/4738758/74319454-48747b00-4d7f-11ea-8c1a-697c848c17c0.png)

So object `UrlScript` contains invalid URL and I can open URL `https://nette.org`, `https://nette.org/` and `https://nette.org/////` too without canonical redirect.

### Steps To Reproduce
Open `https://nette.org/////` for example:

![Snímek obrazovky 2020-02-12 v 10 08 39](https://user-images.githubusercontent.com/4738758/74319688-a5703100-4d7f-11ea-8941-52fa2c3c32a8.png)

Thanks.

**Comment by JanTvrdik** [comment]
There's a built-in path filter in `Nette\Http\RequestFactory` which replaces sequence of `/` with a single `/`. See https://github.com/nette/http/blob/v3.0.3/src/Http/RequestFactory.php#L28

**Comment by janbarasek** [comment]
Yes, of course, but I think this is a natural duplication that should be redirected.

**Comment by janbarasek** [comment]
@dg How can I solve this issue? This bug makes a number of duplicate pages and damages SEO.

I think firstly I must fix `Url` entity which should support sequence of `/` in case of homepage or ending with it.

Thanks.

**Comment by dg** [comment]
This is related to nette/http, I know about it and it will be addressed in version 4.0.
